### PR TITLE
Add on demand flight status support

### PR DESCRIPTION
**Pull request by tsolakoua**

### CODE REVIEW

File: src/main/java/com/amadeus/resources/DatedFlight.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  private @Getter Segments[] segments;
+  private @Getter Legs[] legs;
+
+  @ToString
+  public class FlightDesignator {
+    protected FlightDesignator() {
+    }
+
+    private @Getter String carrierCode;
+    private @Getter int flightNumber;
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/DatedFlight.java:27]: Missing the parameter operationalSuffix String: 
image

File: src/main/java/com/amadeus/resources/DatedFlight.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    private @Getter Departure departure;
+    private @Getter Arrival arrival;
+  }
+
+  @ToString
+  public class Departure {
+    protected Departure() {
+    }
+
+    private @Getter Timings[] timings;
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/DatedFlight.java:45]: Missing gate and terminal: 
image

File: src/main/java/com/amadeus/resources/DatedFlight.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  @ToString
+  public class Arrival {
+    protected Arrival() {
+    }
+
+    private @Getter Timings[] timings;
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/DatedFlight.java:53]: Missing gate and terminal: 

image

File: src/main/java/com/amadeus/resources/DatedFlight.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  @ToString
+  public class Timings {
+    protected Timings() {
+    }
+
+    private @Getter String qualifier;
+    private @Getter String value;
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/DatedFlight.java:62]: Missing delays: 
image

**Review by anthonyroux — CHANGES_REQUESTED** [review]
The resource is not complete, I added some missing elements in comment but I think it needs to be reviewed. For this please use the API reference: https://developers.amadeus.com/self-service/category/air/api-doc/on-demand-flight-status/api-reference and not the API response (that doesn't always contain all the optional fields). 

image

**Comment by tsolakoua** [comment]
@anthonyroux Thanks for the review,  I updated the resource. Let me know if it's ok now.

**Pull request merged**